OsmAnd-java/src/net/osmand/binary/BinaryMapIndexReader.java

```diff
@@ -405,7 +405,12 @@ public String getRegionName() {
 		}
 		String ls = rg.get(0);
 		if (ls.lastIndexOf('_') != -1) {
-			return ls.substring(0, ls.lastIndexOf('_')).replace('_', ' ');
+			if (!ls.endsWith(".obf") || !ls.matches(".*\\d+.*")) {
+				return ls.substring(0, ls.lastIndexOf('_')).replace('_', ' ');
+			} else if (ls.matches("[a-zA-Z_-]+([0-9]+_*){3}[.a-z]+")){
```

Comment:
Please remove regex and such ambiguous numbers as "13" ? The code should be easy to read & verify now is totally cumbersome.